corehq/apps/hqcase/management/commands/reindex_es_native.py

```diff
@@ -0,0 +1,141 @@
+        )
+        parser.add_argument(
+            "--monitor-task-id",
+            help="Pass in the task ID of an existing reindex to monitor it's progress. This will not kick off"
+                 "a new reindex."
+        )
+
+    def handle(self, source_index, target_index, **options):
+        es = get_es_export()
+        if not es.indices.exists(source_index):
+            raise CommandError(f"Source index does not exist: '{source_index}'")
+
+        target_index_info = ES_META[target_index]
+        target_index = target_index_info.index
+        _initialize_target(es, target_index_info)
+
+        source_count = _get_doc_count(es, source_index)
+        target_count = _get_doc_count(es, target_index)
+        print("Number of docs in source index", source_count)
+        print("Number of docs in target index", target_count)
+
+        if options.get("print_index_size"):
+            return
+
+        elif source_count == target_count:
+            print("Doc counts are same, nothing left to reindex. Exiting!")
+            return
+
+        if not options["monitor_task_id"]:
+            print(f"Starting reindex for index from '{source_index}' to '{target_index}'")
+            task_id = start_reindex(es, source_index, target_index)
+        else:
+            task_id = options["monitor_task_id"]
+        check_task_progress(es, task_id)
+
+        print(inspect.cleandoc(f"""
+            If you would like to check, you may run this
+            command again with --print_index_size to make sure
+            doc counts match across both indices.
+
+            Once you are satisfied that the reindex is complete you should run the following:
+
+            ./manage.py initialize_es_indices --index {target_index} --set-for-usage
+        """))
+
+
+def _initialize_target(es, target_index_info):
+    if es.indices.exists(target_index_info.index):
+        print(f"Target index '{target_index_info.index}' already exists. Skipping initialization.")
+        return
+
+    print("Creating target index")
+    initialize_index(es, target_index_info)
+    set_index_reindex_settings(es, target_index_info.index)
+
+    print("Setting number of replicas to 0")
+    es.indices.put_settings({
+        "index.number_of_replicas": 0
+    }, index=target_index_info.index)
+
+
+def start_reindex(es, source_index, target_index):
+    reindex_query = {
+        "source": {
+            "index": source_index,
+        },
+        "dest": {
+            "index": target_index
+        },
+        "conflicts": "proceed"
+    }
+
+    result = es.reindex(reindex_query, wait_for_completion=False, request_timeout=300)
+    task_id = result["task"]
+    return task_id
+
+
+def check_task_progress(es, task_id):
+    node_id = task_id.split(':')[0]
+    node_name = es.nodes.info(node_id, metric="name")["nodes"][node_id]["name"]
+    print(f"Task with ID '{task_id}' running on '{node_name}'")
+    while True:
```

Comment:
If the task finishes (or fails?), a task won't be found. May be that needs to be handled here.

Reply:
it is handled: https://github.com/dimagi/commcare-hq/pull/29828/files#diff-841873087305d51a77852a28d8ed9ff3f281c65032f5e4c8d8196f237d19f837R136-R142